plutus-core/plutus-ir/src/PlutusIR/Transform/Inline/Inline.hs

```diff
@@ -190,27 +187,41 @@ processTerm = handleTerm <=< traverseOf termSubtypes applyTypeSubstitution where
             t' <- processTerm t
             -- Use 'mkLet': we're using lists of bindings rather than NonEmpty since we might
             -- actually have got rid of all of them!
-            pure $ mkLet a NonRec bs' t'
+            pure $ mkLet ann NonRec bs' t'
         -- This includes recursive let terms, we don't even consider inlining them at the moment
-        t -> forMOf termSubterms t processTerm
-    applyTypeSubstitution :: Type tyname uni a -> InlineM tyname name uni fun a (Type tyname uni a)
-    applyTypeSubstitution t = gets isTypeSubstEmpty >>= \case
-        -- The type substitution is very often empty, and there are lots of types in the program,
-        -- so this saves a lot of work (determined from profiling)
-        True -> pure t
-        _    -> typeSubstTyNamesM substTyName t
-    -- See Note [Renaming strategy]
-    substTyName :: tyname -> InlineM tyname name uni fun a (Maybe (Type tyname uni a))
-    substTyName tyname = gets (lookupType tyname) >>= traverse liftDupable
-    -- See Note [Renaming strategy]
-    substName :: name -> InlineM tyname name uni fun a (Maybe (Term tyname name uni fun a))
-    substName name = gets (lookupTerm name) >>= traverse renameTerm
-    -- See Note [Inlining approach and 'Secrets of the GHC Inliner']
-    -- Already processed term, just rename and put it in, don't do any further optimization here.
-    renameTerm ::
-        InlineTerm tyname name uni fun a
-        -> InlineM tyname name uni fun a (Term tyname name uni fun a)
-    renameTerm (Done t) = liftDupable t
+        t -> do
+            -- process all subterms
+            processedT <- forMOf termSubterms t processTerm
+            -- consider call site inlining for each node that have been unconditionally inlined
```

Comment:
I think you mean "have _not_ been unconditionally inlined"

Reply:
hm...I meant that have gone through the unconditional inliner. I'll revise that.